### PR TITLE
Server cache working

### PR DESCRIPTION
**Pull request by Georgehe4**
@devoncarew @lukechurch 
- Files cached locally for 10 minutes (default) - can be changed, please comment if you have suggestions.
- Files moved to static subdirectories.
- Sample up at http://dev.dartproject-1000.appspot.com/.
- Still need to cache codemirror javascript somehow - a bit different from the other files since it's in a package.

screen shot 2015-08-16 at 9 44 09 pm

#650

### CODE REVIEW

File: web/scripts/embed.dart [hunk 1]
```diff
@@ -4,8 +4,8 @@
 
 // Currently not in use
 
-import 'package:dart_pad/embed/embed.dart' as embed;
-import 'package:dart_pad/polymer/polymer.dart';
+import '../../lib/embed/embed.dart' as embed;
+import '../../lib/polymer/polymer.dart';
```
Comment (lukechurch) [inline web/scripts/embed.dart]: I'm not quite sure why we've made this change?

**Comment by lukechurch** [comment]
Thanks @Georgehe4 this looks like a good thing. I'm correct in thinking this only change the embedded version? The main dartpad site is still serving statics out of the root folder with 'no-cache' set? Don't need to fix that in this PR, but if you could file a tracking bug that'd be great.

File: web/scripts/embed.dart (hunk 1 above)
Comment (Georgehe4) [inline web/scripts/embed.dart]: We need to update pub to find the correct packages within the scripts directory. 
The change was auto-generated when I moved the files around into the script subdirectory.

**Comment by Georgehe4** [comment]
Correct, only embedded/mobile versions will use files that are cached.
We may also want to cache codemirror - this would mean that the main dartpad site will use cached files.

File: web/scripts/embed.dart (hunk 1 above)
Comment (lukechurch) [inline web/scripts/embed.dart]: Given that files are in lib, and haven't moved, I would have thought that the old references would still have worked after running pub get. However, @devoncarew knows a lot more about how this piece of machinery works that I do. Lets get his opinion on the right thing to do here.

Comment (Georgehe4) [inline web/scripts/embed.dart]: Looks like a pub update fixes the issue. I'll resolve the changes accordingly.

**Comment by Georgehe4** [comment]
There are some dependency issues with pub. I'll look at it tomorrow.

File: web/scripts/embed.dart (hunk 1 above)
Comment (devoncarew) [inline web/scripts/embed.dart]: I suspect these changes were made by the analysis server's refactoring code? Agree that we want to prefer the `package:` references.

**Comment by devoncarew** [comment]
Wow, this PR has a huge performance benefit:

screen shot 2015-08-17 at 12 45 36 am

I have some ideas for caching codemirror (in a future PR).

File: web/scripts/embed.dart (hunk 1 above)
Comment (devoncarew) [inline web/scripts/embed.dart]: Looking at this, we may be able to keep embed.dart in the same folder (`web/`) and just move its compiled output. Let's talk in person.

**Comment by devoncarew** [comment]
lgtm!

**Comment by Georgehe4** [comment]
Merging.

**Pull request merged**